### PR TITLE
feat: add bucket sort aggregation

### PR DESCRIPTION
**Pull request by adrianboratyn**

### CODE REVIEW

File: src/lib/aggregations/get-bucket-sort.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    bucket_sort: BucketSortAggregationBody
+}
+
+export type BucketSort = {
+    [bucketPath: string]: { order: Order }
+}
+
+// note: all field are optional, because
+// you can use bucket sort for both sorting and truncating result buckets
+// but you also can use it for only sorting or only truncating
```
Comment (ada-krupa) [inline src/lib/aggregations/get-bucket-sort.ts:13]: maybe `@description` instead?

**Review by przemyslawwalczak — APPROVED** [review]

File: src/lib/aggregations/get-bucket-sort.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export type BucketSortAggregationBody = {
+    sort?: Array<BucketSort>
+    from?: number
+    size?: number
+}
+
+export const getBucketSortAggregation = (paramters: BucketSortAggregationBody): BucketSortAggregation => ({
```
Comment (ada-krupa) [inline src/lib/aggregations/get-bucket-sort.ts:20]: parameters

File: src/lib/aggregations/get-bucket-sort.ts (hunk 1 above)
Comment (adrianboratyn) [reply src/lib/aggregations/get-bucket-sort.ts:13]: alright

Comment (przemyslawwalczak) [reply src/lib/aggregations/get-bucket-sort.ts:13]: in that case the `parameters` should be also optional 🙏 (defaulting to empty `{}` if undefined)

**Review by ada-krupa — APPROVED** [review]

**Pull request merged**